tools/ci_build/github/azure-pipelines/templates/telemetry-steps.yml

```
@@ -3,7 +3,8 @@
 
 steps:
   - powershell: |
-      if ($env:TELEMETRYGUID) {
+      $telemetry_guid = ${{ variables['TELEMETRYGUID'] }}
```

`${{ variables['TELEMETRYGUID'] }}` is interpreted by ADO pipeline. It will be replaced to real values before this script start to run.

---

does it expand to a quoted string? e.g. `"guid"` instead of `guid`. if so, that seems odd enough to document somewhere...

---

maybe we can also be explicit about the variable type
```suggestion
      [string]$telemetry_guid = ${{ variables['TELEMETRYGUID'] }}
```

---

The value of the var has quotes inside when I set it.

---

Let me see if I should add extra quotes here.